src/Raven.Server/ServerWide/Commands/CompareExchangeCommands.cs

```diff
@@ -333,6 +334,9 @@ public class AddOrUpdateCompareExchangeCommand : CompareExchangeCommandBase
         public AddOrUpdateCompareExchangeCommand(string database, string key, BlittableJsonReaderObject value, long index, JsonOperationContext contextToReturnResult, string uniqueRequestId, bool fromBackup = false)
             : base(database, key, index, contextToReturnResult, uniqueRequestId, fromBackup)
         {
+            if (value?.TryGetMember(Constants.CompareExchange.ObjectFieldName, out object _) != true)
+                throw new CompareExchangeInvalidValueFormatException($"Invalid format for the compare exchange value of '{key}', the value expected to be wrapped in '{Constants.CompareExchange.ObjectFieldName}' property, but actually it is {value}");
```

Comment:
what is the implication for old atomic guards?

Reply:
Hmm good point... 
The problematic scenario is when some of the nodes already committed this with the old code, while for other nodes it is only appended and not committed.

Reply:
so either schema upgrade, or removing this check